connexion/validators/form_data.py

```diff
@@ -1,85 +1,57 @@
 import logging
 import typing as t
 
-from jsonschema import Draft4Validator, ValidationError, draft4_format_checker
-from starlette.datastructures import FormData, Headers, UploadFile
+from jsonschema import ValidationError, draft4_format_checker
+from starlette.datastructures import Headers, UploadFile
 from starlette.formparsers import FormParser, MultiPartParser
-from starlette.types import Receive, Scope
+from starlette.types import Scope
 
 from connexion.exceptions import BadRequestProblem, ExtraParameterProblem
-from connexion.json_schema import Draft4RequestValidator
+from connexion.json_schema import Draft4RequestValidator, format_error_with_path
 from connexion.uri_parsing import AbstractURIParser
-from connexion.utils import is_null
+from connexion.validators import AbstractRequestBodyValidator
 
 logger = logging.getLogger("connexion.validators.form_data")
 
 
-class FormDataValidator:
+class FormDataValidator(AbstractRequestBodyValidator):
     """Request body validator for form content types."""
 
     def __init__(
         self,
-        scope: Scope,
-        receive: Receive,
         *,
         schema: dict,
-        validator: t.Type[Draft4Validator] = None,
         required=False,
         nullable=False,
         encoding: str,
-        uri_parser: t.Optional[AbstractURIParser] = None,
         strict_validation: bool,
+        uri_parser: t.Optional[AbstractURIParser] = None,
```

Comment:
Yes, true for the case of `application/x-www-form-urlencoded` but for `multipart/form-data` the parameters are sent in different parts instead of query string.
But indeed, not necessarily as surprising as I thought (URI <> body) :)